src/main/java/com/minecolonies/core/client/gui/WindowHutAllInventory.java

```diff
@@ -125,23 +126,28 @@ private void locate(final Button button)
 
         for (BlockPos blockPos : containerList)
         {
-            final BlockEntity rack = Minecraft.getInstance().level.getBlockEntity(blockPos);
-            if (rack instanceof TileEntityRack)
+            final BlockEntity entity = Minecraft.getInstance().level.getBlockEntity(blockPos);
+            Optional<IStorageBlockInterface> storageInterface = ModStorageBlocks.getStorageBlockInterface(entity);
```

Comment:
If you want to use optionals, I'd recommend actually using the optional chaining. Otherwise we prefer using nullables.
So make this
```java
int count = ModStorageBlocks.getStorageBlockInterface(entity).map(storageBlock -> storageBlock.getCount(storage.getItemStack(), storage.ignoreDamageValue(), false)).orElse(0);
```

Reply:
Switched to nullable vs optional chaining where appropriate.